src/components/ReactTable/InfiniteTable.tsx

```diff
@@ -0,0 +1,198 @@
+          >
+            {row.cells.map(cell => (
+              <Td
+                {...cell.getCellProps()}
+                // Header can be () => null or a string, only use data-label if it's a string
+                {...(typeof cell.column.Header === 'string'
+                  ? { 'data-label': cell.column.Header }
+                  : undefined)}
+                display={cell.column.display}
+                key={cell.column.id}
+              >
+                {cell.render('Cell')}
+              </Td>
+            ))}
+          </Tr>
+          {!!renderSubComponent && row.isExpanded ? (
+            <Tr className='expanded-details'>
+              <Td colSpan={visibleColumns.length} style={tdStyle}>
+                {renderSubComponent(row)}
+              </Td>
+            </Tr>
+          ) : null}
+        </Fragment>
+      )
+    })
+  }, [
+    rows,
+    prepareRow,
+    rowDataTestKey,
+    rowDataTestPrefix,
+    onRowClick,
+    renderSubComponent,
+    visibleColumns.length,
+  ])
+
+  return (
+    <InfiniteScroll
+      hasMore={hasMore}
+      next={loadMore}
+      dataLength={data.length}
+      loader={loader}
+      scrollableTarget={scrollableTarget}
+    >
+      <Table ref={tableRef} variant={variant} size={tableSize} {...getTableProps()}>
+        {displayHeaders && (
+          <Thead>
+            {headerGroups.map(headerGroup => (
+              <Tr {...headerGroup.getHeaderGroupProps()}>
+                {headerGroup.headers.map(column => (
+                  <Th
+                    {...column.getHeaderProps(column.getSortByToggleProps())}
+                    color='text.subtle'
+                    textAlign={column.textAlign}
+                    display={column.display}
+                    // we need to pass an arg here, so we need an anonymous function wrapper
+                    // eslint-disable-next-line react-memo/require-usememo
+                    _hover={{ color: column.canSort ? hoverColor : 'text.subtle' }}
+                  >
+                    <Flex justifyContent={column.justifyContent} alignItems={column.alignItems}>
+                      {column.render('Header')}
+                      <Flex alignItems='center'>
+                        {column.isSorted ? (
+                          column.isSortedDesc ? (
+                            <ArrowDownIcon
+                              ml={2}
+                              aria-label={translate('common.table.sortedDesc')}
+                            />
+                          ) : (
+                            <ArrowUpIcon ml={2} aria-label={translate('common.table.sortedAsc')} />
+                          )
+                        ) : null}
+                      </Flex>
+                    </Flex>
+                  </Th>
+                ))}
+              </Tr>
+            ))}
+          </Thead>
+        )}
+        <Tbody {...getTableBodyProps()}>{renderRows}</Tbody>
```

Comment:
question on this pattern of calling a function to generate props, should we be memoizing them since it's a new object?

Reply:
Can maybe tackle this in a follow up performance PR? Since this is what exists in production right now. All the examples of react-table do it this way. I think it has something to do with if anything changes the entire table needs to re-render for sorting, filtering etc.